api/src/main/java/org/jboss/marshalling/Javax2JakartaClassNameTransformer.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.marshalling;
+
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opalka</a>
+ */
+final class Javax2JakartaClassNameTransformer implements ClassNameTransformer {
+
+    private static final String MAPPING_FILE = "javax2jakarta.mapping";
+    private static final String JAVAX_PREFIX = "javax.";
+    private static final String JAKARTA_PREFIX = "jakarta.";
+    private static final String[] javaxPackageNames;
+    private static final String[] jakartaPackageNames;
+
+    static {
+        try {
+            final Properties props = new Properties();
+            props.load(ClassNameTransformer.class.getResourceAsStream(MAPPING_FILE));
+            final Set<String> javaxPackages = props.stringPropertyNames();
+            final int count = javaxPackages.size();
+            javaxPackageNames = new String[count];
+            jakartaPackageNames = new String[count];
+            int i = 0;
+            for (String javaxPackageName : javaxPackages) {
+                javaxPackageNames[i] = javaxPackageName;
+                jakartaPackageNames[i] = props.getProperty(javaxPackageName);
+                i++;
+            }
+        } catch (final Throwable t) {
+            throw new RuntimeException(t);
+        }
+    }
+
+    @Override
+    public String transformInput(final String className) {
+        if (className.startsWith(JAVAX_PREFIX)) {
+            for (int i = 0; i < javaxPackageNames.length; i++) {
```

Comment:
A linear search here might be unnecessarily costly.  A binary search over a sorted array might give better results.

Reply:
Yes, or maybe have the javax package names sorted in a [][] array. That way, you already know that className starts with javax., so you test the character at position 6, and you go to index [characterFound-'a'] in that array, this would save some processing time, because you know that, for example, having found an 'x', you end up in an array that contains only three options to test.

Reply:
Fixed.